game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -337,6 +346,68 @@ public int getDefaultUnitsStackSize() {
     return Math.max(0, Integer.parseInt(stack));
   }
 
+  /** Returns the unit color associated with the player named {@code playerName}. */
+  public Optional<Color> getUnitColor(final String playerName) {
+    // already loaded, just return
+    if (unitColors.containsKey(playerName)) {
```

Comment:
Properties seems to store data in memory in its own map, is there a benefit to the `unitColors` cache?

Reply:
Honestly, I'm not sure. This is how its done for `playerColors` so I followed the same pattern. Gonna consider examining that further out of scope for this PR.

Reply:
I strongly suspect the caching is a cache on top of a cache already. I think the issue is the original "read from map.properties" comment was misleading/not correct, leading to the extra caching.

Not ideal to keep as is, would be a bit simpler without the cache -  but okay.. We can leave it for another day. A TODO comment indicating this uncertainty in caching would be good to see, but not strictly necessary.